### DISCUSSION TITLE
Generate all classes without any modifiers or variants

### DISCUSSION THREAD

**Discussion by andyjamesn** [tailwindlabs/tailwindcss#10218] [Help]
I have worked with Tailwind enough to understand that version 3 being fully JIT it is not practical to try pre generate classes as there are just so many options and the possibilities are near endless.

However I have a use case where I need to generate a list of all possible classes without any modifiers or variants.

So p-1, p-2, text-center etc only. No hover:p-2 or other modifiers or variants will ever be needed.

I need it to create a token select of possible classes for a UI I am building. The variant is added to the class after it is selected hence why I do not need the variants.

**Comment by wongjn** [comment]
You could use [a regular expression in `safelist`](https://tailwindcss.com/docs/content-configuration#using-regular-expressions), like:
```js
module.exports = {
  // …
  safelist: [{ pattern: /^/ }],
  // …
}
```

**Reply by andyjamesn** [reply]
Thanks. I know how to do it with the safelist. But my use case is different. I do not need it for generating the css. I purely want to output a list of all classes only that I can use in a UI.. I do not need to generate the styles

**Reply by wongjn** [reply]
You could try to emulate what this test does:
https://github.com/tailwindlabs/tailwindcss/blob/7b3de616e9c26211c605d2378b9ceb4888a8cfa1/tests/getClassList.test.js#L4-L28

**Reply by andyjamesn** [reply]
Great find! That worked perfectly.

**Comment by Jakeduncan00** [comment]
// tailwind.config.js
module.exports = {
  variants: {
    extend: {
      backgroundColor: ['active'],
      // ...
      borderColor: ['focus-visible', 'first'],
      // ...
      textColor: ['visited'],
    }
  },
}


[TigerIsHome
](https://www.tigerishome.org/)
